plugins/lookup/conjur_variable.py

```diff
@@ -263,43 +325,60 @@ def _store_secret_in_file(value):
 
     return [secrets_file.name]
 
-
 class LookupModule(LookupBase):
 
     def run(self, terms, variables=None, **kwargs):
         if terms == []:
             raise AnsibleError("Invalid secret path: no secret path provided.")
         elif not terms[0] or terms[0].isspace():
             raise AnsibleError("Invalid secret path: empty secret path not accepted.")
+        
+        # We should register the variables as LookupModule options.
+        #
+        # Doing this has some nice advantages if we're considering supporting
+        # a set of Ansible variables that could sometimes replace environment
+        # variables.
+        #
+        # Registering the variables as options forces them to adhere to the
+        # behavior described in the DOCUMENTATION variable. An option can have
+        # both a Ansible variable and environment variable source, which means
+        # Ansible will do some juggling on our behalf.
+        self.set_options(var_options=variables, direct=kwargs)
+
+        appliance_url = self.get_var_value("conjur_appliance_url")
+        account = self.get_var_value("conjur_account")
+        authn_login = self.get_var_value("conjur_authn_login")
+        authn_api_key = self.get_var_value("conjur_authn_api_key")
+        cert_file = self.get_var_value("conjur_cert_file")
+        authn_token_file = self.get_var_value("conjur_authn_token_file")
```

Comment:
I think we have some unexpected behavior here that breaks backwards compatibility.

Previously, we had a section in the `DOCUMENTATION` variable that described the `conjur_authn_token_file` variable, but it was being accessed exclusively as an environment variable, so the specified default value was not being applied. Now that we're accessing the value this way, the default value is being applied when we would expect it not to.

We either need to add some special handling for this variable to maintain backwards compatibility, or release this as part of a major version bump.

Reply:
In the past, the authn_token_file value was generated by either an environmental variable or the default path (/var/run/conjur/access-token).

The newest modifications allow for the authn_token_file value to be pulled from either the environmental variable, extra variables, or the default path (/var/run/conjur/access-token).

The behavior was the same as earlier, but we have updated only the ini name of authn_token_file.

Reply:
> In the past, the authn_token_file value was generated by either an environmental variable _**or the default path (/var/run/conjur/access-token)**_

The last part of this isn't true - previously, the lookup plugin [checked the `CONJUR_AUTHN_TOKEN_FILE` environment variable](https://github.com/cyberark/ansible-conjur-collection/blob/cb3faccdc51d3c8e8f031480f110c3a0e038abbf/plugins/lookup/conjur_variable.py#L300-L302), and if didn't exist, no default value was applied, and instead the plugin [retrieved a new authn token](https://github.com/cyberark/ansible-conjur-collection/pull/203/files#diff-bbe5a68ac6b3a3f3cdb55039bafcc0b3dccf14ed5a234f7562bfe4264d6a2331R412).

Check out this PR's [Jenkins build](https://jenkins.conjur.net/blue/organizations/jenkins/cyberark--ansible-conjur-collection/detail/extra-vars-poc/5/pipeline), which is failing on the lookup plugin's end-to-end tests.

```
TASK [Retrieve Conjur variable] ************************************************
fatal: [localhost]: FAILED! => {"msg": "An unhandled exception occurred while templating
  '{{lookup('conjur_variable', 'ansible/test-secret')}}'. Error was a
  <class 'ansible.errors.AnsibleError'>, original message: An unhandled exception
  occurred while running the lookup plugin 'conjur_variable'. Error was a
  <class 'ansible.errors.AnsibleError'>, original message: Conjur authn token
  file `/var/run/conjur/access-token` was not found on the host. Conjur authn token
  file `/var/run/conjur/access-token` was not found on the host"}
```

The existing `DOCUMENTATION` variable section describing the `authn_token_file` parameter is inaccurate, but we aren't using `self.get_option`, so it isn't being enforced. Backwards compatibility should be maintained if we updated the new `authn_token_file` section to specify that it doesn't have a default value, and isn't required.

Reply:
To ensure backward compatibility for conjur_authn_token_file. We have removed the default key parameter in the conjur_authn_token_file documentation section and also added an extra check for conjur_authn_token_file. 

**if not variable_value and key != "conjur_authn_token_file":**
             raise AnsibleError("The value of the {0} variable is not set".format(key))
             
The token can either be selected from environmental/extra-vars variables or generated within the code.

Reply:
Thanks, @ramavenkata-loya!